db_dtypes/__init__.py

```diff
@@ -98,7 +98,8 @@ def astype(self, dtype, copy=True):
 
     def __arrow_array__(self, type=None):
         return pyarrow.array(
-            self.to_numpy(), type=type if type is not None else pyarrow.time64("ns"),
+            self.to_numpy(dtype="object"),
```

Comment:
This might have performance implications, but it does seem to prevent the cast to float64 for empty arrays. Also, the `dtype` seems to be `object` whenever there are any values in the array, anyway.

Reply:
Sweet. That seems to have done the trick. I copied the implementation from your PR https://github.com/googleapis/python-db-dtypes-pandas/pull/9/files#diff-1956943f14005805ef968dfc37c26fe3eee995786f62a1c66dee5a29d9b1a251R111